agenta-web/src/lib/helpers/utils.ts

```diff
@@ -26,25 +35,44 @@ export const EvaluationTypeLabels: Record<EvaluationType, string> = {
     [EvaluationType.auto_webhook_test]: "Webhook Test",
 }
 
-export const saveOpenAIKey = (key: string) => {
+export const saveLlmProviderKey = (providerName: string, keyValue: string) => {
     if (typeof window !== "undefined") {
         // TODO: add encryption here
-        localStorage.setItem(openAItoken, key)
+        const keys = JSON.parse(localStorage.getItem(llmAvailableProvidersToken) ?? "{}")
+        keys[providerName] = keyValue
+        localStorage.setItem(llmAvailableProvidersToken, JSON.stringify(keys))
     }
 }
 
-export const getOpenAIKey = (): string => {
+export const getLlmProviderKeys = () => {
     // precedence order: local storage, env variable, empty string
-    let key
+    let keys: {[key: string]: string} = {}
+    for (const prov of llmAvailableProviders) {
+        keys[prov] = ""
+    }
+
+    if (typeof window !== "undefined") {
+        keys = JSON.parse(localStorage.getItem(llmAvailableProvidersToken) ?? "{}")
+    }
+    // the "NEXT_PUBLIC_OPENIA_API_KEY logic here must be rethought
```

Comment:
one way to incorporate env vars is to make `llmAvailableProviders` an array of object of shape `{[key: string; title: string]}[]`.

And then use the env like: `process.env.NEXT_PUBLIC_LLM_KEY_${key}`

This would work once the [PR](https://github.com/Agenta-AI/agenta/pull/769) of accessing dynamic env vars is merged.

Reply:
I believe that instead of doing like this, it would be better to do an single object like this: ``{"title": "key"}``
What do you think?

If the ``{[key: string; title: string]}[]`` shape is better, I'll change it right away.

Reply:
Done. Take a look and tell me if I need to change anything